### PR TITLE
enhance(edit): add capture helper for hb templates

### PR DESCRIPTION
**Pull request by kevinslin**
enhance(edit): add capture helper for hb templates

This helper lets users capture parts of a text. Docs: https://github.com/dendronhq/dendron-site/pull/544

### CODE REVIEW

File: packages/common-server/src/template.ts [hunk 1]
```diff
@@ -121,6 +121,14 @@ class TemplateHelpers {
       const day = date.getDay();
       return day;
     },
+
+    capture: (text: string, pattern: string): string => {
```
Comment (jonathanyeung) [inline packages/common-server/src/template.ts:125]: Let's also modify `genTrackPayload` in this file to get data for this helper method.

File: packages/engine-test-utils/src/__tests__/common-all/template.spec.ts [hunk 2]
```diff
@@ -129,6 +129,41 @@ describe(`WHEN running applyTemplate tests`, () => {
       });
     });
 
+    describe("AND WHEN using capture helper", () => {
```
Comment (jonathanyeung) [inline packages/engine-test-utils/src/__tests__/common-all/template.spec.ts:132]: very nitpick: `AND` usually combos with the THEN (the verification clause, as in doing multiple verifications) as opposed to the WHEN.

**Review by jonathanyeung — CHANGES_REQUESTED** [review]

**Comment by jonathanyeung** [comment]
lgtm, but looks like there's a cli test error.

**Comment by kevinslin** [comment]
> lgtm, but looks like there's a cli test error.

fixed

**Pull request merged**